src/main/java/uk/gov/hmcts/darts/common/service/impl/CaseCommonServiceImpl.java

```diff
@@ -25,16 +25,20 @@ public class CaseCommonServiceImpl implements CaseCommonService {
     public CourtCaseEntity retrieveOrCreateCase(String courthouseName, String caseNumber, UserAccountEntity userAccount) {
         String courthouseNameUC = StringUtils.toRootUpperCase(courthouseName);
         Optional<CourtCaseEntity> foundCase = caseRepository.findByCaseNumberAndCourthouse_CourthouseName(caseNumber, courthouseNameUC);
-        return foundCase.map(entity -> setCourtCaseLastDateModifiedBy(entity, userAccount))
-            .orElseGet(() -> createCase(courthouseName, caseNumber, userAccount));
+        return foundCase
+            .map(entity -> setCourtCaseLastDateModifiedBy(entity, userAccount))
+            .orElseGet(() -> createCase(courthouseName, caseNumber, userAccount))
+            .validateIsExpired();
```

Comment:
As per other comment, we should keep this validation out of common code that fetches the case.

Reply:
Updated